client/state/directly/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+import {
+	initialize as initializeDirectly,
+	askQuestion as askDirectlyQuestion,
+	maximize as maximizeDirectly,
+	minimize as minimizeDirectly,
+	openAskForm as openDirectlyAskForm,
+} from 'lib/directly';
+import {
+	DIRECTLY_ASK_QUESTION,
+	DIRECTLY_INITIALIZE,
+	DIRECTLY_MAXIMIZE,
+	DIRECTLY_MINIMIZE,
+	DIRECTLY_OPEN_ASK_FORM,
+} from 'state/action-types';
+
+export const askQuestion = ( questionText, name, email ) => ( dispatch ) => {
+	dispatch( { type: DIRECTLY_ASK_QUESTION, questionText, name, email } );
+	askDirectlyQuestion( questionText, name, email );
+};
```

Comment:
Ok — so is there a rule of thumb for when to use middleware instead of a thunk from an action creator for side-effecting operations? Many action creators initiate side-effects in their returned thunk functions (e.g. [savePost](https://github.com/Automattic/wp-calypso/blob/master/client/state/posts/actions.js#L193-L217)) so what exactly makes that case different from my Directly side-effecting operations?

I think intuitively middleware feels like the right place to me too, but I could convince myself either way so I'd love some more concrete principles to ground myself on.